common/src/main/java/com/microsoft/identity/common/exception/IntuneAppProtectionPolicyRequiredException.java

```diff
@@ -45,6 +60,128 @@ public IntuneAppProtectionPolicyRequiredException(final String errorCode,
+
+        final String upn = (originalParameters.getAccountManagerAccount() != null) ?
+                originalParameters.getAccountManagerAccount().name :
+                originalParameters.getLoginHint();
+
+        setAccountUpn(upn);
+
+        String uId = originalParameters.getLocalAccountId();
+
+        if (TextUtils.isEmpty(uId)) {
+            com.microsoft.identity.common.internal.logging.Logger.info(TAG, "Local account id is empty, attempting get user id from home account id");
+            uId = getUIdFromHomeAccountId(
+                    originalParameters.getHomeAccountId()
+            );
+        }
+
+        setAccountUserId(uId);
+
+        final Authority authority = originalParameters.getAuthority();
+        setAuthorityUrl(authority.getAuthorityURL().toString());
+
+        final String homeAccountId = originalParameters.getHomeAccountId();
+        String tenantId = null;
+
+        if (homeAccountId != null) {
+            final Pair<String, String> tenantInfo = StringUtil.getTenantInfo(homeAccountId);
+            tenantId = tenantInfo.second;
+        }
+
+        if (TextUtils.isEmpty(tenantId) && authority instanceof AzureActiveDirectoryAuthority) {
+            tenantId = ((AzureActiveDirectoryAuthority) authority).mAudience.getTenantId();
+
+        }
+        setTenantId(tenantId);
+    }
+
+    public IntuneAppProtectionPolicyRequiredException(final String errorCode,
+                                                      final String errorMessage,
+                                                      final BrokerSilentTokenCommandParameters originalParameters) {
+        super(errorCode, errorMessage, null);
+
+        final String upn = (originalParameters.getAccountManagerAccount() != null) ?
+                originalParameters.getAccountManagerAccount().name :
+                originalParameters.getLoginHint();
+
+        setAccountUpn(upn);
+
+        String uId = originalParameters.getLocalAccountId();
+
+        if (TextUtils.isEmpty(uId)) {
+            com.microsoft.identity.common.internal.logging.Logger.info(TAG, "Local account id is empty, attempting get user id from home account id");
+            uId = getUIdFromHomeAccountId(
+                    originalParameters.getHomeAccountId()
+            );
+        }
+
+        setAccountUserId(uId);
+
+        final Authority authority = originalParameters.getAuthority();
+        setAuthorityUrl(authority.getAuthorityURL().toString());
+
+        final String homeAccountId = originalParameters.getHomeAccountId();
+        String tenantId = null;
+
+        if (homeAccountId != null) {
+            final Pair<String, String> tenantInfo = StringUtil.getTenantInfo(homeAccountId);
+            tenantId = tenantInfo.second;
+        }
+
+        if (TextUtils.isEmpty(tenantId) && authority instanceof AzureActiveDirectoryAuthority) {
+            tenantId = ((AzureActiveDirectoryAuthority) authority).mAudience.getTenantId();
+
```

Comment:
If it is possible for tenantId toned up being empty, log that.

In the interest brevity, maybe hold all these logging tasks until the end, and log a single-line summary of what is not available.